### PR TITLE
Reapply "Add source file name for template instantiations in -ftime-trace"

### PR DESCRIPTION
**Pull request by usx95**
Fix the Windows test.

### CODE REVIEW

**Comment by ilya-biryukov** [comment]
This still fails on Windows, see the buildkite errors.
I believe the `filename(..., style::posix)` does not do what you intend, it actually assumes the inputs are in posix format, which is not the case on Windows. You want to use the native style

File: clang/unittests/Support/TimeProfilerTest.cpp [hunk 1]
```diff
@@ -60,13 +75,29 @@ bool compileFromString(StringRef Code, StringRef Standard, StringRef FileName) {
...
 
+std::string GetMetadata(json::Object *Event) {
+  std::string Metadata;
+  llvm::raw_string_ostream OS(Metadata);
+  if (json::Object *Args = Event->getObject("args")) {
+    if (auto Detail = Args->getString("detail"))
+      OS << Detail;
+    if (auto File = Args->getString("file"))
+      OS << ", "
+         << llvm::sys::path::filename(File->str(),
```
Comment (ilya-biryukov) [inline clang/unittests/Support/TimeProfilerTest.cpp:86]: Could you add a comment that explains that we do this to make the tests outputs the same between Windows and Linux?

Comment (usx95) [reply clang/unittests/Support/TimeProfilerTest.cpp]: Done.

**Comment by usx95** [comment]
presubmits are passing now.

**Review by ilya-biryukov — APPROVED** [review]
LGTM, thanks

**Comment by vitalybuka** [comment]
It introduces another issue: https://lab.llvm.org/buildbot/#/builders/164/builds/1246/steps/14/logs/stdio

**Review by overmighty — COMMENTED** [review]
Why was the file a-abfdec1d.o.tmp committed?

**Pull request merged**